api/keyconjurer/serverless_functions.go

```diff
@@ -188,18 +190,13 @@ func (h *Handler) GetTemporaryCredentialEventHandler(ctx context.Context, req *e
 	}
 
 	log = h.log.WithFields(logrus.Fields{"username": event.Credentials.Username, "idp": event.AuthenticationProvider, "account_id": event.AppID})
-	provider, ok := h.authenticationProviders.Get(event.AuthenticationProvider)
+	provider, ok := providers.Get(event.AuthenticationProvider)
 	if !ok {
 		log.Infof("unknown provider %q", provider)
 		return ErrorResponse(ErrCodeInvalidProvider, "invalid provider")
 	}
 
-	if _, err := provider.Authenticate(ctx, event.Credentials); err != nil {
-		log.Errorf("failed to authenticate user: %s", err)
-		return ErrorResponse(ErrCodeInvalidCredentials, "credentials are incorrect")
-	}
-
-	response, err := provider.GenerateSAMLAssertion(ctx, event.Credentials, event.AppID)
+	response, err := provider.GenerateSAMLAssertion(ctx, providers.Credentials(event.Credentials), event.AppID)
```

Comment:
The `Authenticate()` call that was previously here was used to validate that a users credentials were valid. This was a duplication of effort because `GenerateSAMLAssertion` _always_ verified the users credentials. As a result, a call to `Authenticate()` was an unnecessary duplication of effort that involves a round-trip to Okta, which we'd like to avoid.

Reply:
this is great!